### ISSUE TITLE
mb_strpos(): Offset not contained in string

### ISSUE THREAD

**Issue by mprofitlich** [symphonists/search_index#65]
I sometimes got the  PHP error _mb_strpos(): Offset not contained in string_. I cannot find out when it happens and when not.

There’s an old [thread from 2012 on the Symphony forum](http://www.getsymphony.com/discuss/thread/46163/15/#position-286) in which Nikolai Shmakov gives a solution that also works for me:

Replace line 383 in class.search_index_php
`$p = $match[0][1];`
against 
`$p = $match[0][1]/2; `

**Comment by animaux** [comment]
I think I used that too some time ago. But if I remember correctly, the side-effect was that the extracts were offset from the actual keywords.

**Comment by mprofitlich** [comment]
That’s true. The results are still correct but the extract shown often doesn’t show the keyword (which is there but at another place in the text). How did you solve or circumvent the problem?

**Comment by animaux** [comment]
`¯\_(ツ)_/¯`
